code/game/machinery/firealarm.dm

```diff
@@ -419,12 +496,22 @@
 			return TRUE
 	return FALSE
 
+// Taking melee damage always triggers the alarm if panel is open
+/obj/machinery/firealarm/attacked_by(obj/item/attacking_item, mob/living/user, list/modifiers, list/attack_modifiers)
+	. = ..()
+	if(!. || !panel_open || buildstage != FIRE_ALARM_BUILD_SECURED)
+		return
+	alarm()
+
+// Taking any damage has a rng chance of triggering the alarm regardless of panel state
 /obj/machinery/firealarm/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir)
 	. = ..()
-	if(.) //damage received
-		if(atom_integrity > 0 && !(machine_stat & BROKEN) && buildstage != FIRE_ALARM_BUILD_NO_CIRCUIT)
-			if(prob(33) && buildstage == FIRE_ALARM_BUILD_SECURED) //require fully wired electronics to set of the alarms
-				alarm()
+	if(!.) //damage received
+		return
```

Comment:
this comment seems the same as before negating the check
should this comment be negated too?